codecarbon/emissions_tracker.py

```diff
@@ -370,38 +372,35 @@ def _init_output_methods(self, api_key):
         """
         Prepare the different output methods
         """
-        self.persistence_objs: List[BaseOutput] = list()
-
         if self._save_to_file:
-            self.persistence_objs.append(
+            self._output_handlers.append(
                 FileOutput(
                     os.path.join(self._output_dir, self._output_file),
```

Comment:
Twice the `output_dir`. Maybe we can give as input of FileOutput:

- `output_dir`
-  `output_file_name`
- `on_csv_write`

WDYT?

Reply:
Done